src/common/libflux/message.h

```diff
@@ -56,26 +57,41 @@ struct flux_match {
     const char *topic_glob;       /* glob matching topic string (or NULL) */
 };
 
-#define FLUX_MATCH_ANY (struct flux_match){ \
-    .typemask = FLUX_MSGTYPE_ANY, \
-    .matchtag = FLUX_MATCHTAG_NONE, \
-    .topic_glob = NULL, \
+static inline struct flux_match flux_match_create (int typemask,
+                                                   uint32_t matchtag,
+                                                   const char *topic_glob)
+{
+    struct flux_match m = {typemask, matchtag, topic_glob};
+    return m;
 }
-#define FLUX_MATCH_EVENT (struct flux_match){ \
-    .typemask = FLUX_MSGTYPE_EVENT, \
-    .matchtag = FLUX_MATCHTAG_NONE, \
-    .topic_glob = NULL, \
-}
-#define FLUX_MATCH_REQUEST (struct flux_match){ \
-    .typemask = FLUX_MSGTYPE_REQUEST, \
-    .matchtag = FLUX_MATCHTAG_NONE, \
-    .topic_glob = NULL, \
-}
-#define FLUX_MATCH_RESPONSE (struct flux_match){ \
-    .typemask = FLUX_MSGTYPE_RESPONSE, \
-    .matchtag = FLUX_MATCHTAG_NONE, \
-    .topic_glob = NULL, \
+static inline void flux_match_destroy (struct flux_match m)
+{
+    if (m.topic_glob) {
```

Comment:
Test for non-null is not needed.

Reply:
Quite so, I shouldn't have put that in there, not sure why I did.